schema/token.graphql

```diff
@@ -0,0 +1,384 @@
+  benefits: [Benefit!] @derivedFrom(field: "token")
+
+  "note from creator to member interested in joining the whitelist"
+  whitelistApplicantNote: String
+
+  "link for creator to member interested in joining the whitelist"
+  whitelistApplicantLink: String
+
+  "holders having some amount of this token"
+  accounts: [TokenAccount!] @derivedFrom(field: "token")
+
+  "number of accounts to avoid aggregate COUNT"
+  accountsNum: Int!
+
+  "number of completed Amm Sales"
+  ammNonce: Int!
+
+  "number of revenue shares issued"
+  revenueShareNonce: Int!
+
+  "whether it has been deissued or not"
+  deissued: Boolean!
+}
+
+type TokenChannel @entity @index(fields: ["token", "channel"], unique: true) {
+  "tokenId ++ channelId"
+  id: ID!
+
+  "token reference"
+  token: Token! @unique
+
+  "channel reference"
+  channel: Channel! @unique
+}
+
+type RevenueShare @entity {
+  "tokenId ++ runtime id for revenue split"
+  id: ID!
+
+  "token which this revenue share is for"
+  token: Token!
+
+  "list of participating members"
+  stakers: [RevenueShareParticipation!] @derivedFrom(field: "revenueShare")
+
+  "block at which the revenue share was issued"
+  createdIn: Int!
+
+  "starting block"
+  startingAt: Int!
+
+  "duration"
+  duration: Int!
+
+  "ending block"
+  endsAt: Int!
+
+  "Number of participant to aggregating COUNT on stackers"
+  participantsNum: Int!
+
+  "total number of HAPI allocated"
+  allocation: BigInt!
+
+  "amounts claimed so far in order to avoid SUM aggregations"
+  claimed: BigInt!
+
+  "finalized"
+  finalized: Boolean!
+}
+
+type Benefit @entity {
+  "counter"
+  id: ID!
+
+  "token the benefit is for"
+  token: Token!
+
+  "emoji for the benefit"
+  emojiCode: String
+
+  "title for the benefit"
+  title: String!
+
+  "description for the benefit"
+  description: String!
+
+  "order in which the benefits is displayed"
+  displayOrder: Int!
+}
+
+type Sale @entity @index(fields: ["token", "createdIn"], unique: true) {
+  "runtime TokenId ++ runtime sale nonce"
+  id: ID!
+
+  "token for which the sale is issued"
+  token: Token!
+
+  "constant price in HAPI for each token"
+  pricePerUnit: BigInt!
+
+  "total amount of token on sale"
+  tokenSaleAllocation: BigInt!
+
+  "amount of tokens that has been sold"
+  tokensSold: BigInt!
+
+  "creation block"
+  createdIn: Int!
+
+  "sale start block"
+  startBlock: Int!
+
+  "sale duration"
+  durationInBlocks: Int!
+
+  "ending block"
+  endsAt: Int!
+
+  "terms and conditions text"
+  termsAndConditions: String!
+
+  "max amount that member can purchase"
+  maxAmountPerMember: BigInt
+
+  "transactions references"
+  transactions: [SaleTransaction!] @derivedFrom(field: "sale")
+
+  "finalized"
+  finalized: Boolean!
+
+  "vesting schedule for sale"
+  vestedSale: VestedSale @derivedFrom(field: "sale")
+
+  "sale funds source"
+  fundsSourceAccount: TokenAccount!
+}
+
+type VestedSale @entity @index(fields: ["sale", "vesting"], unique: true) {
```

Comment:
Vesting schedules id is `cliffBlock++vestingduration++cliffPercent` they are referenced by 
- account (this can happen during initial allocation, sale and possibly with amm sales (as discussed on discord))
- sales

So there could be a vesting schedule that is not referenced by any sale and there could also be a sale that does not reference any vesting. So it was a optional-to-optional (not possible in subsquid) relationship that I have broken down into
two many-to-one for Sale, VestingSchedule towards VestedSale